### PR TITLE
feat(js): Introduce a Tooltip primitive

### PR DESCRIPTION
**Pull request by desiprisg**
### Screenshots

Screenshot 2024-07-31 at 9 21 37 AM

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: packages/js/src/ui/components/primitives/Dropdown/DropdownRoot.tsx [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { ComponentProps } from 'solid-js';
+import { Popover } from '../Popover';
+
+export const DropdownRoot = (props: ComponentProps<typeof Popover.Root>) => {
+  return <Popover.Root placement="bottom" fallbackPlacements={['top']} {...props} />;
```
Comment (desiprisg) [inline packages/js/src/ui/components/primitives/Dropdown/DropdownRoot.tsx:5]: Add a dropdown specific default.

File: packages/js/src/ui/components/primitives/Tooltip/TooltipContent.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import { JSX, onCleanup, onMount, Show, splitProps } from 'solid-js';
+import { Portal } from 'solid-js/web';
+import { AppearanceKey, useFocusManager } from '../../../context';
+import { useStyle } from '../../../helpers';
+import { Root } from '../../elements';
+import { useTooltip } from './TooltipRoot';
+
+export const tooltipContentVariants = () =>
+  'nt-bg-neutral-alpha-900 nt-p-4 nt-shadow-[0px_5px_20px_0px_rgba(0,0,0,0.20)] nt-z-10 nt-rounded-lg nt-text-background nt-text-sm';
```
Comment (desiprisg) [inline packages/js/src/ui/components/primitives/Tooltip/TooltipContent.tsx]: Using `colorNeutral` and `background` here to get the inverted  colors. We could swap `colorNeutral` with `foreground`, wdyt?

File: packages/js/src/ui/components/Notification/DefaultNotification.tsx [hunk 3]
```diff
@@ -58,26 +59,54 @@ export const DefaultNotification = (props: DefaultNotificationProps) => {
...
-              </Button>
+              <Tooltip.Root>
+                <Tooltip.Trigger
+                  asChild={(props) => (
+                    <Button appearanceKey="notificationRead__button" size="icon" variant="icon" {...props}>
+                      <ReadAll />
+                    </Button>
+                  )}
+                />
+                <Tooltip.Content>{t('notification.actions.read.toolTip')}</Tooltip.Content>
```
Comment (LetItRock) [inline packages/js/src/ui/components/Notification/DefaultNotification.tsx:70]: ✏ **suggestion:** maybe we should add an additionally `whitespace-nowrap`

File: packages/js/src/ui/components/primitives/Tooltip/TooltipContent.tsx (hunk 2 above)
Comment (LetItRock) [reply packages/js/src/ui/components/primitives/Tooltip/TooltipContent.tsx]: it looks identical to me, but semantically I think that `foreground` is better suited

File: packages/js/src/ui/components/primitives/Tooltip/TooltipContent.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    onCleanup(() => {
+      removeActive(floatingEl!);
+    });
+  });
+
+  return (
+    <div
+      ref={setFloating}
+      //id is necessary here because this is a portal
```
Comment (LetItRock) [inline packages/js/src/ui/components/primitives/Tooltip/TooltipContent.tsx]: ❓ **question:** is something missing?

File: packages/js/src/ui/components/primitives/Tooltip/TooltipRoot.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    fallbackValue: false,
+  });
+
+  const onClose = () => {
+    setIsOpen(false);
+  };
+
+  const onToggle = () => {
+    setIsOpen((prev) => !prev);
+  };
```
Comment (LetItRock) [inline packages/js/src/ui/components/primitives/Tooltip/TooltipRoot.tsx]: ✏ **suggestion:** leftover?

**Review by LetItRock — APPROVED** [review]
awesome! 🙌

File: packages/js/src/ui/components/primitives/Tooltip/TooltipRoot.tsx (hunk 5 above)
Comment (desiprisg) [reply packages/js/src/ui/components/primitives/Tooltip/TooltipRoot.tsx]: yeap, thanks!

File: packages/js/src/ui/components/primitives/Tooltip/TooltipContent.tsx (hunk 4 above)
Comment (desiprisg) [reply packages/js/src/ui/components/primitives/Tooltip/TooltipContent.tsx]: Ah, no. It's a leftover comment in the popover as well.

**Pull request merged**